src/main/java/org/thoughtslive/jenkins/plugins/jira/service/JiraEndPoints.java

```diff
@@ -49,7 +50,8 @@
   Call<Object> createIssue(@Body Object issue);
 
   @PUT("rest/api/2/issue/{issueIdOrKey}")
-  Call<Object> updateIssue(@Path("issueIdOrKey") String issueIdOrKey, @Body Object issue);
+  Call<Object> updateIssue(@Path("issueIdOrKey") String issueIdOrKey, @Body Object issue,
+      @QueryMap Map<String, String> queryMap);
```

Comment:
Can we try to add this queryMap to every step or I would appreciate it if you would like a log a follow-up JIRA to extend this further to all the steps.

Reply:
I think it would be better to log a follow up JIRA ticket to extend it to all steps. The scope of this PR has already expanded significantly.

Reply:
Created the JIRA ticket for extending to all steps: https://issues.jenkins-ci.org/browse/JENKINS-60776

Reply:
Thank you for logging this follow up JIRA.